pkg/controller/configmap/config.go

```diff
@@ -623,6 +621,24 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 		if !needChange.agent && inSlice(sec, operatortypes.AgentSections) {
 			needChange.agent = true
 		}
+		if !needChange.agent && inSlice(sec, []string{"k8s", "coe"}) {
+			if hasSection(prevCfg, sec) != hasSection(currCfg, sec) {
```

Comment:
In most cases, k8s can code sections are compulsory, so previous config and current config will have both.

Reply:
> In most cases, k8s can code sections are compulsory, so previous config and current config will have both.

But we still need a check here as the section could be `coe`?

Reply:
Yes, agreed, if adding or deleting a section, like coe or k8s, restarting node agent might be necessary. More precisely,  we can check if added or deleted section has key from AgentRestartKey or not.